go/buildhelper/node.go

```diff
@@ -0,0 +1,207 @@
+package buildhelper
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+	"strings"
+	"sync"
+
+	"github.com/ethereum/go-ethereum"
+
+	"github.com/obscuronet/obscuro-playground/go/buildhelper/buildconstants"
+	"github.com/obscuronet/obscuro-playground/go/buildhelper/helpertypes"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/obscuronet/obscuro-playground/go/log"
+	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
+	"github.com/obscuronet/obscuro-playground/go/obscuronode/nodecommon"
+)
+
+var (
+	nonceLock sync.RWMutex
+)
+
+type EthNode struct {
+	port      uint
+	ipaddress string
+	apiClient *EthAPI
+	id        common.Address
+}
+
+func NewEthNode(id common.Address, ipaddress string, port uint) (obscurocommon.L1Node, error) {
```

Comment:
This needs a cleanup - You're correct, obscuro doesn't need the ID, but we're using the ID for the validation steps.

Let me add a TODO on it 👍  I think changing this might require a bit of work on the simulation itself.